yt/frontends/boxlib/data_structures.py

```diff
@@ -1650,11 +1650,10 @@ def _parse_parameter_file(self):
                     self.parameters[l[0].strip()] = l[1].strip()
 
         # set the periodicity based on the integer BC runtime parameters
-        is_periodic = self.parameters["geometry.is_periodic"].split()
-        periodicity = [bool(val) for val in is_periodic]
-        for _ in range(self.dimensionality, 3):
-            periodicity += [True]  # pad to 3D
-        self.periodicity = ensure_tuple(periodicity)
+        self.periodicity[:] = False
```

Comment:
```
    self.periodicity[:] = False
TypeError: 'tuple' object does not support item assignment
```

Reply:
Yup, apparently in this particular class we're forcing the tuple type, but there's also as note that says it should be fixed...
https://github.com/yt-project/yt/blob/9138a0a7330e642e040f8dc0a459f7dba347fa05/yt/frontends/boxlib/data_structures.py#L624

Anyway, this is a different problem than the one you're fixing, sorry again... I'll fix this in a separate branch and if all goes well we'll be able to merge this one right after.

Reply:
So it's much worse than I anticipated, and there is a lot of inconsistency right now in how we handle this particular attribute across frontends. You may simply disregard this and backpedal to using tuples. I'm sorry for the back and forth, thanks for your reactivity.

Reply:
I think I will set it to `True` as other readers in the frontend and ignore it altogether. This will cause the least breakage.